extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -58,26 +66,36 @@ export const popupMessageListener = (request: Request) => {
     mnemonicPhrase,
     password,
     keyPair,
+    imported = false,
   }: {
     mnemonicPhrase: string;
     password: string;
     keyPair: KeyPair;
+    imported?: boolean;
   }) => {
     const { publicKey, privateKey } = keyPair;
 
     const allAccounts = allAccountsSelector(store.getState());
+    const accountName = `Account ${allAccounts.length + 1}`;
 
     store.dispatch(
       logIn({
         publicKey,
         mnemonicPhrase,
-        allAccounts: [...allAccounts, publicKey],
+        allAccounts: [
+          ...allAccounts,
+          {
+            publicKey,
+            name: accountName,
+            imported,
+          },
+        ],
       }),
     );
 
     const keyMetadata = {
       key: {
-        extra: { mnemonicPhrase },
+        extra: { imported, mnemonicPhrase },
```

Comment:
this somewhat tangentially related: we need to differentiate between imported/generated accounts and include that info in allAccounts.